io/net/infiniband/ip_over_ib.py

```diff
@@ -0,0 +1,204 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM
+# Author: Prudhvi Miryala<mprudhvi@linux.vnet.ibm.com>
+# Ip Over IB test
+# IPoIB can run over two infiniband transports
+# Unreliable Datagram (UD) mode or Connected mode (CM)
+
+
+import os
+import time
+import netifaces
+from avocado import main
+from avocado import Test
+from avocado.utils.software_manager import SoftwareManager
+from avocado.utils import build
+from avocado.utils import archive
+from avocado.utils import process
+from avocado.utils import git
+
+
+class ip_over_ib(Test):
+    '''
+    Ip Over IB Test
+    IPoIB can run over two infiniband transports,
+    Unreliable Datagram (UD) mode or Connected mode (CM)
+    '''
+    def setUp(self):
+        '''
+        To check and install dependencies for the test
+        '''
+        sm = SoftwareManager()
+        depends = ["openssh-clients", "gcc"]
+        for pkg in depends:
+            if not sm.check_installed(pkg) and not sm.install(pkg):
+                self.skip("%s package is need to test" % pkg)
+        interfaces = netifaces.interfaces()
+        self.IF = self.params.get("Iface", default="")
+        self.PEER_IP = self.params.get("PEERIP", default="")
+        if self.IF not in interfaces:
+            self.skip("%s interface is not available" % self.IF)
+        if self.PEER_IP == "":
+            self.skip("%s peer machine is not available" % self.PEER_IP)
+        self.to = self.params.get("timeout", default="600")
+        self.IPERF_RUN = self.params.get("IPERF_RUN", default="0")
+        self.NETSERVER_RUN = self.params.get("NETSERVER_RUN", default="0")
+        self.iper = os.path.join(self.srcdir, 'perf')
+        self.netperf = os.path.join(self.srcdir, 'netperf')
+        tarball = self.fetch_asset('ftp://ftp.netperf.org/netperf/'
+                                   'netperf-2.7.0.tar.bz2')
+        archive.extract(tarball, self.netperf)
+        version = os.path.basename(tarball.split('.tar.')[0])
+        self.neperf = os.path.join(self.netperf, version)
+        tmp = "scp -r %s root@%s:" % (self.neperf, self.PEER_IP)
+        if process.system(tmp, shell=True, ignore_status=True) != 0:
+            self.skip("unable to copy the netperf into peer machine")
+        tmp = "cd /root/netperf-2.7.0;./configure ppc64le;make"
```

Comment:
to run iperf and netperf root access need, sry for not mentioned in README file. i will update README file